### ISSUE TITLE
Unrecognized VM option error when adding --dev flag on Windows

### ISSUE THREAD

**Issue by laumacirule** [jruby/jruby#8322]
**Environment Information**
This issue is reproducible in windows-latest Github actions runner:
* OS Name: Microsoft Windows Server 2022 Datacenter
System Type: x64-based PC
Processor(s): 1 Processor(s) Installed: AMD64 Family 25 Model 1 Stepping 1 AuthenticAMD ~2445 Mhz
* Java version
openjdk version "1.8.0_412"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_4[12](https://github.com/eazybi/eazybi/actions/runs/9385148880/job/25842570302#step:5:13)-b08)
OpenJDK 64-Bit Server VM (Temurin)(build 25.412-b08, mixed mode)
* Environment variables
JAVA_OPTS: -Xmx2048m
JRUBY_OPTS: --dev -W0
JRUBY_VERSION: 9.3.13.0

**Expected Behavior**
Set the JAVA_OPTS provided by setting the `--dev` flag as it happens on Ubuntu runners.

**Actual Behavior**
The options provided by the `--dev` flag seem to be passed as one string giving the following error
```
rspec spec
Unrecognized VM option 'TieredCompilation
-XX:TieredStopAtLevel=1
-XX:-PrintWarnings
-Djruby.compile.mode=OFF
-Djruby.compile.invokedynamic=false
-Djnr.ffi.asm.enabled=false
-Xverify:none'
```

**Comment by headius** [comment]
I believe this came up in another discussion but I can't find it right now. Basically, the quoting of these options on Windows does not appear to work properly anymore.

**Comment by headius** [comment]
I believe this must be a bug in the native launcher on Windows.

**Comment by headius** [comment]
Hmm, the options appear to be pushed separately by the launcher, based on this code:

https://github.com/jruby/jruby-launcher/blob/7366acf974255c68fc55b6a1f600362005a0eec1/argparser.cpp#L338-L342

@laumacirule Can you provide a bit more context here... how was this whole job launched? Something like `bundle exec` or similar? Running at a command line with `--dev` should work fine... perhaps you can confirm that. I am thinking this is happening when calling a subprocess and trying to pass the same JVM options through.

**Comment by laumacirule** [comment]
Yes, you're right. It failed when running `bundle exec`, but the step before, where a rake task for db migrations was executed, was OK:
```
bin/rake db:migrate
shell: C:\msys64\usr\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
env:
  JAVA_OPTS: -Xmx2048m
  JRUBY_OPTS: --dev -J-Xmx2048m -W0
  JRUBY_VERSION: 9.3.13.0
...

bundle exec rspec spec/models/dwh_spec.rb 
  shell: C:\msys64\usr\bin\bash.EXE --noprofile --norc -e -o pipefail {0}
  env:
    JAVA_OPTS: -Xmx2048m
    JRUBY_OPTS: --dev -J-Xmx2048m -W0
    JRUBY_VERSION: 9.3.13.0
...
Unrecognized VM option 'TieredCompilation
-XX:TieredStopAtLevel=1
-XX:-PrintWarnings
-Djruby.compile.mode=OFF
-Djruby.compile.invokedynamic=false
-Djnr.ffi.asm.enabled=false
-Xverify:none'
```

**Comment by headius** [comment]
Ok, this may be a bug in how Bundler determines the command-line to `exec` as a subprocess. It's probably taking all those JVM arguments and slamming them together as one, which the JVM (or our launcher on Windows) does not parse properly. Seems like we have enough to reproduce, but I'll have to spin up my Windows environment to be sure.

Just to confirm, you have `--dev` in the `JRUBY_OPTS` env var, yes? And then bundle exec fails this way?

**Comment by laumacirule** [comment]
I ran the tests with the `--dev` option on the GitHub windows-2025 runner and with JRuby 9.4.12.1 and did not see this error anymore. Not sure what changed, but as this is no longer an issue I can reproduce, I'm closing it.
